app/user_cleanup.go

```diff
@@ -0,0 +1,94 @@
+package app
+
+import (
+	"strings"
+
+	"github.com/rancher/rancher/pkg/systemaccount"
+	v3 "github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type userCleanup struct {
+	users    v3.UserInterface
+	clusters v3.ClusterInterface
+	projects v3.ProjectInterface
+}
+
+func cleanupOrphanedSystemUsers(management *config.ManagementContext) {
+	u := userCleanup{
+		users:    management.Management.Users(""),
+		clusters: management.Management.Clusters(""),
+		projects: management.Management.Projects(""),
+	}
+	u.cleanup()
+}
+
+func (u *userCleanup) cleanup() {
+	users, err := u.users.List(v1.ListOptions{})
+	if err != nil {
+		logrus.Errorf("Error listing users during system account users cleanup: %v", err)
+		return
+	}
+	clusters, err := u.clusters.List(v1.ListOptions{})
+	if err != nil {
+		logrus.Errorf("Error listing clusters during system account users cleanup: %v", err)
+		return
+	}
+	for _, user := range users.Items {
+		systemUser := false
+		for _, principal := range user.PrincipalIDs {
+			if strings.HasPrefix(principal, "system://") {
+				systemUser = true
+				break
+			}
+		}
+		if !systemUser {
+			continue
+		}
+
```

Comment:
Maybe would be better to have separate method to check if the systemUser should be deleted, instead of having all logic inline here. And then we could do:
for... {
		if systemUser {
			checkCleanup()
		}
}